source/common/buffer/watermark_buffer.h

```diff
@@ -40,20 +45,25 @@ class WatermarkBuffer : public OwnedImpl {
   uint32_t highWatermark() const { return high_watermark_; }
 
 private:
-  void checkHighWatermark();
+  void checkHighAndOverflowWatermarks();
   void checkLowWatermark();
 
+  Runtime::Loader& runtime_;
```

Comment:
If we snap the multiplier, I think we won't need to store the reference and can just use the free variant to grab the multiplier at construction?

Reply:
Updated; wasn't sure what you meant about the "free variant" - lmk if this isn't what you had in mind.